libs/go/libp2p_node/dht/dhtclient/dhtclient.go

```diff
@@ -881,51 +713,6 @@ func (dhtClient *DHTClient) handleAeaAddressStream(stream network.Stream) {
 		return
 	}
 
-	msg := &acn.AcnMessage{}
-	err = proto.Unmarshal(buf, msg)
-	if err != nil {
-		lerror(err).Str("op", "resolve").Msg("couldn't deserialize acn message")
-		// TOFIX(LR) setting Msgs to err.Error is potentially a security vulnerability
-		status := &acn.StatusBody{
-			Code: acn.ERROR_SERIALIZATION,
-			Msgs: []string{err.Error()},
-		}
-		statusPerformative := &acn.StatusPerformative{Body: status}
-		response := &acn.AcnMessage{
-			Performative: &acn.Status{Status: statusPerformative},
-		}
-		buf, err = proto.Marshal(response)
-		ignore(err)
-		err = utils.WriteBytes(stream, buf)
-		ignore(err)
-		err = stream.Close()
-		ignore(err)
-		return
-	}
-
-	// Get LookupRequest message
-	var lookupRequest *acn.LookupRequestPerformative
-	switch pl := msg.Performative.(type) {
-	case *acn.LookupRequest:
-		lookupRequest = pl.LookupRequest
-	default:
-		err = errors.New("Unexpected payload, expected LookupRequest")
-		status := &acn.StatusBody{Code: acn.ERROR_UNEXPECTED_PAYLOAD, Msgs: []string{err.Error()}}
-		statusPerformative := &acn.StatusPerformative{Body: status}
-		response := &acn.AcnMessage{
-			Performative: &acn.Status{Status: statusPerformative},
-		}
-		buf, err = proto.Marshal(response)
-		ignore(err)
-		err = utils.WriteBytes(stream, buf)
```

Comment:
where is this done now? in utils I guess..worth checking

Reply:
it's done in utils with helper methods readrequest/sendresponse functions